hathor/feature_activation/feature.py

```diff
@@ -0,0 +1,27 @@
+#  Copyright 2023 Hathor Labs
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from enum import Enum, unique
+
+
+@unique
+class Feature(Enum):
+    """
+    An enum containing all features that participate in the feature activation process, past or future,
+    activated or not, for all networks. Features should NOT be removed from this enum, to preserve history.
+    Their values should NOT be changed either.
```

Comment:
Why is this the case? Could you include the reason in this description?

Like, is it used by the storage so it should be preserved so databases are compatible between node versions? Or are the values used inside blocks? Or even on the p2p protocol (which I don't think it's the case)?

Reply:
> Like, is it used by the storage so it should be preserved so databases are compatible between node versions? Or are the values used inside blocks? Or even on the p2p protocol (which I don't think it's the case)?

No, none of those are the case. The reason is just the the value is used to parse criteria from settings. Changing them could mess up with configuration. Ideally I would use `StrEnum` and each option would use `enum.auto()`, so each name and value are equal strings. But it's only available from Python 3.11

Added more docstring info in https://github.com/HathorNetwork/hathor-core/pull/611/commits/44615a281d8be68479e90e818e774ca1887c2a83